src/tyr/serializers.cc

```diff
@@ -106,10 +120,14 @@ json::ArrayPtr waypoints(const valhalla::Trip& trip) {
   const auto& legs = trip.routes(0).legs();
   for (const auto& leg : legs) {
     for (const auto& location : leg.location()) {
-      if (&location == &leg.location(0) && &leg != &*legs.begin()) {
+      const auto& is_first_location = &location == &leg.location(0);
+      if (is_first_location && &leg != &*legs.begin()) {
```

Comment:
i was going to suggest the same thing. instead of doing pointer arithmetic why not just put a bool on the outside of the loop. set it to true and then set it to false inside the loop. then you can keep the for each and not have to change it into a counter based for loop